crates/provider-keyvalue-redis/src/lib.rs

```diff
@@ -141,6 +163,54 @@ impl KvRedisProvider {
         Ok(conn.clone())
     }
 
+    async fn invoke_on_set(&self, target: impl AsRef<str>, bucket: &str, key: &str, value: &str) {
+        let mut cx: async_nats::HeaderMap = async_nats::HeaderMap::new();
+        for (k, v) in
+        wasmcloud_provider_sdk::wasmcloud_tracing::context::TraceContextInjector::default_with_span(
+        )
+        .iter()
+    {
+        cx.insert(k.as_str(), v.as_str())
+    }
+        // send wrpc event to invoke the component exported code using wrpc.invoke
+        let wrpc = get_connection().get_wrpc_client_custom(target.as_ref(), None);
+        let func = "on_set";
+        let _ = wrpc
+            .invoke_values::<_, (&str, &str, &str), ((),)>(
+                Some(cx),
+                target.as_ref(),
+                func,
+                (bucket, key, value),
+                &[[]; 0],
+            )
+            .await;
+        info!("Key set: {}", key);
+    }
+
+    async fn invoke_on_delete(&self, target: impl AsRef<str>, bucket: &str, key: &str) {
+        let mut cx: async_nats::HeaderMap = async_nats::HeaderMap::new();
+        for (k, v) in
+        wasmcloud_provider_sdk::wasmcloud_tracing::context::TraceContextInjector::default_with_span(
+        )
+        .iter()
+    {
+        cx.insert(k.as_str(), v.as_str())
+    }
+        // send wrpc event to invoke the component exported code using wrpc.invoke
+        let wrpc = get_connection().get_wrpc_client_custom(target.as_ref(), None);
+        let func = "on_delete";
+        let _ = wrpc
+            .invoke_values::<_, (&str, &str), ((),)>(
+                Some(cx),
+                target.as_ref(),
+                func,
+                (key, bucket),
+                &[[]; 0],
+            )
+            .await;
```

Comment:
Same comments re: macro from provider SDK and `on_delete` that's generated by wit_bindgen_wrpc